### PR TITLE
[Poppler] Add feature 'qt'

### PR DESCRIPTION
**Pull request by sunderme**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by sunderme** [comment]
This enables the optional feature to build poppler-qt interface which is needed by some programs (e.g. texstudio)

**Comment by sunderme** [comment]
@microsoft-github-policy-service agree

File: ports/poppler/portfile.cmake [hunk 1]
```diff
@@ -38,6 +38,12 @@ endif()
 if("zlib" IN_LIST FEATURES)
     string(APPEND POPPLER_PC_REQUIRES " zlib")
 endif()
+if("qt" IN_LIST FEATURES)
+    list(APPEND FEATURE_OPTIONS "-DENABLE_QT6=ON")
+else()
+    list(APPEND FEATURE_OPTIONS "-DENABLE_QT6=OFF")
+endif()
+
```
Comment (dg0yt) [inline ports/poppler/portfile.cmake:46]: This option should be added to `vcpkg_check_features` instead.

**Review by dg0yt — COMMENTED** [review]
Did you verify that the exported cmake config and pc files are complete with the qt feature enabled?

**Comment by sunderme** [comment]
> Did you verify that the exported cmake config and pc files are complete with the qt feature enabled?

poppler-qt builds and is detected by cmake.

File: ports/poppler/portfile.cmake (hunk 1 above)
Comment (sunderme) [reply ports/poppler/portfile.cmake:46]: ENABLE_QT6 needs to be set explicitely `on` or `off`.
Doing that with `Features`/`Inverted features` seems not to be clearer or shorter, but I can change that.

**Comment by MonicaLiu0311** [comment]
Feature `qt` is tested successfully in the following triplet:
x86-windows
x64-windows
x64-windows-static

**Review by MonicaLiu0311 — DISMISSED** [review]

File: ports/poppler/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/poppler/portfile.cmake:46]: Please change. `vcpkg_check_features` is just for that. It is just one line in the `FEATURES` section, and the function is already here, so it is much clearer.

**Comment by dg0yt** [comment]
Don't merge yet.

**Comment by dg0yt** [comment]
Tested with x64-mingw-static:
- `lib\pkgconfig\poppler-qt6.pc` doesn't carry Qt dependencies. This is hard to fix without Qt6 pc files.
- `unofficial-poppler` CMake config doesn't cover the Qt backend at all.

IMO at least the CMake config should be extended to cover the new backend.

File: versions/p-/poppler.json [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+    {
+      "git-tree": "729340bfc26472ea2d9dfbdba1132a5d7d96c45d",
+      "version": "23.1.0",
+      "port-version": 3
+    },
+    {
+      "git-tree": "ad4348be6efbdf7155263ecddf4d640a26671ede",
+      "version": "23.1.0",
+      "port-version": 2
+    },
```
Comment (MonicaLiu0311) [inline versions/p-/poppler.json:17]: - [ ] Only one version is added to each modified port's versions file.

Thank you for your contribution, please execute in order:
1. Delete redundant records;
2. Modify the "port-version" in `vcpkg.json` to `2`;
3. Run `./vcpkg x-add-version portName --overwrite-version` to update git-tree.



`--overwrite-version`
After the first modification, please run `./vcpkg x-add-version portName` to generate the corresponding json; for subsequent modifications of the same version, please use `./vcpkg x-add-version portName --overwrite-version` to update git-tree.

**Review by MonicaLiu0311 — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
> IMO at least the CMake config should be extended to cover the new backend.

Do you need help with this? (The pattern is given by the existing patch, which probably should be simply extended.)

File: ports/poppler/unofficial-poppler-config.cmake [hunk 3]
```diff
@@ -20,3 +20,4 @@ endif()
 find_dependency(Iconv) # CMake 3.11
```
Comment (dg0yt) [inline ports/poppler/unofficial-poppler-config.cmake:20]:
~~~cmake
if("qt" IN_LIST features)
    find_dependency(Qt6 CONFIG Core GUI)
endif()

File: ports/poppler/unofficial-poppler-config.cmake [hunk 4]
```diff
@@ -20,3 +20,4 @@ endif()
...
 include("${CMAKE_CURRENT_LIST_DIR}/unofficial-poppler-targets.cmake")
 include("${CMAKE_CURRENT_LIST_DIR}/unofficial-poppler-cpp-targets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-poppler-qt6-targets.cmake")
```
Comment (dg0yt) [inline ports/poppler/unofficial-poppler-config.cmake:23]:
~~~cmake
if("qt" IN_LIST features)
    include...
endif()

**Review by dg0yt — COMMENTED** [review]
It looks good, but please make two more adjustments. (I didn't test the actual syntax.)

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**